sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/nullExpressions.scala

```diff
@@ -90,7 +86,12 @@ case class Coalesce(children: Seq[Expression]) extends Expression {
           }
         }
       """
-    }.mkString("\n"))
+    }
+
+    ev.copy(code = s"""
+      ${ev.isNull} = true;
+      ${ev.value} = ${ctx.defaultValue(dataType)};
+      ${ctx.splitExpressions(ctx.INPUT_ROW, evals)}""")
```

Comment:
We can only split the expressions when they are created from a row object.

Reply:
sorry, I don't see which is the problem here. I see that here the row object is `null`, but the goal is to set `ev.isNull` and `ev.value`, which is done.  May you explain me if there is something I am missing? Thanks.

Reply:
`splitExpressions` puts expression codes in individual functions. Only if those expressions's input is from a row object, we can do this. If their input is from `currentVars`, the splitting doesn't work.

Reply:
It won't cause problem in fact. In that case, it works as before, i.e., `expressions.mkString("\n")`.

Reply:
This is the reason why I put `ev.isNull` and `ev.value` as attributes of the generated class, in this way they can be used as before in the individual functions. If you want, I can try and use the other overloaded method `splitExpressions` without passing the input row to it.

Reply:
Ok. nvm. looks good.